File: 프로그래머스/2/87946. 피로도/피로도.py
from itertools import permutations
def solution(k, dungeons):
    answer = -1
    result = set()
    l = permutations(dungeons,len(dungeons))
    for i in l:
        total = 0
        k2 = k
        for j in i:
            if k2 >= j[0]:
                k2 -= j[1]
                total +=1
            else:
                break
        result.add(total)
    return max(result)

File: 프로그래머스/2/87946. 피로도/피로도.java
import java.util.*;
class Solution {
    public int[] output;
    public boolean[] visited;
    int n,r,c;
    int answer = -1;
    public int solution(int k, int[][] dungeons) {
        n = dungeons.length;
        r = n;
        c = k;
        output = new int[n];
        visited = new boolean[n];
        perm(0,dungeons);
        return answer;
    }
    
    public void perm(int cnt, int[][] dungeons) {
        if(cnt == r) {
            int current = c;
            int count = 0;
            for (Integer i : output) {
                if (dungeons[i][0] <= current) {
                    count +=1;
                    current -= dungeons[i][1];
                } else {
                    break;
                }
            }
            answer = Math.max(answer,count);
            return;
        }
        for (int i=0;i<n;i++){
            if(visited[i]) continue;
            output[cnt] = i;
            visited[i] = true;
            perm(cnt+1,dungeons);
            visited[i] = false;
        }  
    }
}